Kitodo/src/main/java/de/sub/goobi/helper/GoobiScript.java

```diff
@@ -817,35 +817,36 @@ public void deleteTiffHeaderFile(List<Process> processes) {
 
     /**
      * Reset image path in the metadata (possibly delete existing ones first).
-     * 
+     *
      * @param processes
      *            list of Process objects
      */
     public void updateImagePath(List<Process> processes) {
         for (Process proz : processes) {
             try {
-                Fileformat myRdf = serviceManager.getProcessService().readMetadataFile(proz);
-                MetadataType mdt = UghHelper.getMetadataType(proz, "pathimagefiles");
-                List<? extends ugh.dl.Metadata> allImagePaths = myRdf.getDigitalDocument().getPhysicalDocStruct()
+                FileformatInterface myRdf = serviceManager.getProcessService().readMetadataFile(proz);
+                MetadataTypeInterface mdt = UghHelper.getMetadataType(proz, "pathimagefiles");
+                List<? extends MetadataInterface> allImagePaths = myRdf.getDigitalDocument().getPhysicalDocStruct()
                         .getAllMetadataByType(mdt);
                 if (allImagePaths.size() > 0) {
-                    for (Metadata md : allImagePaths) {
+                    for (MetadataInterface md : allImagePaths) {
                         myRdf.getDigitalDocument().getPhysicalDocStruct().getAllMetadata().remove(md);
                     }
                 }
-                Metadata newmd = new Metadata(mdt);
+                MetadataInterface newmd = UghImplementation.INSTANCE.createMetadata(mdt);
                 if (SystemUtils.IS_OS_WINDOWS) {
-                    newmd.setValue("file:/" + serviceManager.getFileService().getImagesDirectory(proz) + proz.getTitle()
-                            + DIRECTORY_SUFFIX);
+                    newmd.setStringValue("file:/" + serviceManager.getFileService().getImagesDirectory(proz)
+                            + proz.getTitle() + DIRECTORY_SUFFIX);
                 } else {
-                    newmd.setValue("file://" + serviceManager.getFileService().getImagesDirectory(proz)
+                    newmd.setStringValue("file://" + serviceManager.getFileService().getImagesDirectory(proz)
                             + proz.getTitle() + DIRECTORY_SUFFIX);
                 }
                 myRdf.getDigitalDocument().getPhysicalDocStruct().addMetadata(newmd);
                 serviceManager.getFileService().writeMetadataFile(myRdf, proz);
                 Helper.setMeldung("kitodoScriptfield", "ImagePath updated: ", proz.getTitle());
 
-            } catch (DocStructHasNoTypeException | UghHelperException | MetadataTypeNotAllowedException e) {
+            } catch (org.kitodo.api.ugh.exceptions.DocStructHasNoTypeException | UghHelperException
```

Comment:
Please do not use full name of the used exception if the short name work too.

Reply:
This was inserted during refactoring, when classes partly still used the equally-named exceptions from the UGH package, and partly from the API package. In such cases, the refactoring tools insert the qualified form automatically. (It is actually necessary at that point.) I had overseen to remove it.